android/plugins/network/src/main/java/com/facebook/flipper/plugins/network/FlipperOkhttpInterceptor.java

```diff
@@ -7,46 +7,83 @@
 
 package com.facebook.flipper.plugins.network;
 
+import android.text.TextUtils;
+import com.facebook.flipper.core.*;
+import com.facebook.flipper.plugins.common.BufferingFlipperPlugin;
 import com.facebook.flipper.plugins.network.NetworkReporter.RequestInfo;
 import com.facebook.flipper.plugins.network.NetworkReporter.ResponseInfo;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
 import javax.annotation.Nullable;
-import okhttp3.Headers;
-import okhttp3.Interceptor;
-import okhttp3.Request;
-import okhttp3.Response;
-import okhttp3.ResponseBody;
+import okhttp3.*;
 import okio.Buffer;
 import okio.BufferedSource;
 
-public class FlipperOkhttpInterceptor implements Interceptor {
+public class FlipperOkhttpInterceptor
+    implements Interceptor, BufferingFlipperPlugin.ConnectionListener {
 
   public @Nullable NetworkFlipperPlugin plugin;
 
+  private Map<String, ResponseInfo> mockResponeMap = new HashMap<>();
+
   public FlipperOkhttpInterceptor() {
     this.plugin = null;
   }
 
   public FlipperOkhttpInterceptor(NetworkFlipperPlugin plugin) {
     this.plugin = plugin;
+    this.plugin.setConnectionListener(this);
+  }
+
+  protected void registerMockResponse(String requestUrl, String method, ResponseInfo response) {
+    mockResponeMap.put(requestUrl + "|" + method, response);
```

Comment:
Is there a way of avoiding string concatenation here and turning this into a proper type? This seems rather fragile could break quite easily.